pkg/reconciler/source/duck/duck.go

```diff
@@ -235,7 +232,7 @@ func (r *Reconciler) computeDiff(current []v1beta2.EventType, expected []v1beta2
 		if c, ok := currentMap[keyFromEventType(&e)]; !ok {
 			toCreate = append(toCreate, e)
 		} else {
-			if !equality.Semantic.DeepEqual(e.Spec, c.Spec) {
+			if !equality.Semantic.DeepDerivative(e.Spec, c.Spec) {
```

Comment:
```suggestion
			if !equality.Semantic.DeepDerivative(c.Spec, e.Spec) {
```

Iirc, the problem was that the "current" eventtypes (variable `c`) were having no namespace in the reference field, so since `DeepDerivative` ignored unset fields in he first variable passed in I _think_ the order should be flipped here

Reply:
hmm, how is that possible?

expected is the one we need to use as a "reference", the current one can have more fields set and we want to ignore them.

Reply:
Looks like you're correct, I tried this locally and everything is fine! Sorry for the confusion